core/src/main/resources/error/error-classes.json

```diff
@@ -290,6 +290,46 @@
           "Null typed values cannot be used as arguments of <functionName>."
         ]
       },
+      "NUM_FORMAT_CONT_THOUSANDS_SEPS" : {
```

Comment:
How is this related to DATA_TYPE_MISMATCH?
Shouldn't this be a separate error_class:
INVALID_NUMBER_FORMAT.*

Reply:
Theoretically, a data type is a set of possible values and a set of allowed operations on it. The number format is not an arbitrary string, it must be in some particular format. So, the error indicates violation of datatype range of values.

Reply:
I think we abuse type check in a few expressions... Some are query compile time static checks.

Reply:
> Some are query compile time static checks

I would like to propose to do refactoring later. Let's assign unique error classes to the errors that come from `checkInputDataTypes()`.

Reply:
@cloud-fan Are you ok with such approach?

Reply:
This must be fixed before we cut 12.0.
Also there is no joy in having an error class with dozens of subclasses because it makes for a deeper doc tree.

By this logic a CONSTRAINT violation is a data type mismatch and so is a NOT NULL violation....

Reply:
> This must be fixed before we cut 12.0.

Definitely, it will be fixed before Spark 12.